channeldb/graph.go

```diff
@@ -1304,6 +1312,46 @@ func putLightningNode(nodeBucket *bolt.Bucket, aliasBucket *bolt.Bucket, node *L
 		return err
 	}
 
+	if err := node.Features.Encode(&b); err != nil {
+		return err
+	}
+
+	numAddresses := uint16(len(node.Addresses))
+	byteOrder.PutUint16(scratch[:2], numAddresses)
+	if _, err := b.Write(scratch[:2]); err != nil {
+		return err
+	}
+
+	for _, address := range node.Addresses {
+		if address.Network() == "tcp" {
+			if address.(*net.TCPAddr).IP.To4() != nil {
+				scratch[0] = 0
+				if _, err := b.Write(scratch[:1]); err != nil {
+					return err
+				}
+
+				copy(scratch[:4], address.(*net.TCPAddr).IP.To4())
+				if _, err := b.Write(scratch[:4]); err != nil {
+					return err
+				}
+			} else {
+				scratch[0] = 1
+				if _, err := b.Write(scratch[:1]); err != nil {
+					return err
+				}
+				copy(scratch[:], address.(*net.TCPAddr).IP.To16())
```

Comment:
Similar comment here about the copy overriding the `addrType` set as the first variable.